null mx record check

bmcculley opened the pull request:
To address issue #10 this adds the ability to check for a null MX record.
It also adds a new error code NULL_MX_RECORD.

michaelherold commented on pyisemail/validators/dns_validator.py:
```diff
@@ -37,8 +37,15 @@ def is_valid(self, domain, diagnose=False):
+            # value. If it's value is 0 it should return null MX.
+            if len(records) > 0:
+                for rdata in records:
+                    if rdata.preference == 0:
```
**issue:** Don't we need to check for a zero length (i.e. `.`) record in addition to the zero preference? My reading of [Section 3 of the RFC](https://www.rfc-editor.org/rfc/rfc7505.html#section-3) says yes. Particularly combined with [the SMTP fallback behavior](https://datatracker.ietf.org/doc/html/rfc5321#section-5.1) of, "If an empty list of MXs is returned, the address is treated as if it was associated with an implicit MX RR, with a preference of 0, pointing to that host," and [the examples at the end of RFC 974](https://datatracker.ietf.org/doc/html/rfc974), it seems like zero does not imply by itself that it's a null MX record.

A test of the behavior would be good to show that it works as intended.

michaelherold reviewed (changes requested):
We definitely need test coverage for this. I'm pretty sure the implementation is incomplete as well.

Please take a look at my comments below.

bmcculley commented:
Thank you @michaelherold after reading the RFC I agree with your comments above. Further it is now my understanding that a Null MX record should also only consist of a single MX record with 0 preference and an empty label. 

I have updated the code accordingly and added some tests to verify that it works as intended. Please review and let me know your thoughts.

michaelherold reviewed (approved):
This is good work, thanks! CI is failing due to an old version of Black not having the proper constraints. I will rebase and get this merged and released!

michaelherold commented on pyisemail/diagnosis/dns_diagnosis.py:
```diff
@@ -21,4 +21,5 @@ class DNSDiagnosis(BaseDiagnosis):
             "Couldn't find an MX record for this domain " "but an A record does exist."
         ),
         "NO_RECORD": "Couldn't find an MX record or A record for this domain.",
+        "NULL_RECORD": "Domain does not support email service."
```
There's a small issue here that I will fix when rebasing. This needs to match the key above.
